WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionSummaryViewModel.cs

```diff
@@ -36,8 +43,39 @@ public TransactionSummaryViewModel(TransactionPreviewViewModel parent, Wallet wa
 		AddressText = info.Destination.ToString();
 		PayJoinUrl = info.PayJoinClient?.PaymentUrl.AbsoluteUri;
 		IsPayJoin = PayJoinUrl is not null;
+
+		var selectCoinsInteraction = SelectCoinsInteraction(walletViewModel);
+
+		SelectCoinsCommand = ReactiveCommand.CreateFromObservable(() => selectCoinsInteraction.Handle(info));
+		SelectCoinsCommand.Do(
+			list =>
+			{
+				info.Coins = list;
+
+				// UpdateTransactionSummary here, but how??
+			}).Subscribe();
```

Comment:
No, you should not touch `TransactionSummaryViewModel` at all. You don't even need `Interaction`. Your dialog should return a set of coins, which will be stored in `TransactionInfo` and the transaction preview screen is responsible for re-building the transaction with the new coins.